hw3/01-characters.js

```diff
@@ -1,4 +1,69 @@
-// url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
+// GET api/characters
+// [
+//     {
+//       "id": 0,
+//       "firstName": "string",
+//       "lastName": "string",
+//       "fullName": "string",
+//       "title": "string",
+//       "family": "string",
+//       "image": "string",
+//       "imageUrl": "string"
+//     }
+//   ]
 
-fetch(url);
+async function addCharacters(characters) {
+  characters.forEach((character) => {
+    const img = document.createElement('img');
+    img.src = character.imageUrl;
+    Object.assign(img.style, { width: '165px', height: '195px' });
+    img.alt = `Portrait of ${character.fullName}`;
+
+    const personName = document.createElement('h2');
+    personName.textContent = character.fullName;
+    personName.className = 'person-name mt-3 fw-bold text-center mx-auto w-75';
+    personName.style.fontSize = '24px';
+
+    const personTitle = document.createElement('h3');
+    personTitle.textContent = character.title;
+    personTitle.className = 'person-title mt-1 fw-bold text-center mx-auto w-100';
+    personTitle.style.fontSize = '16px';
+
+    const characterCard = document.createElement('div');
+    characterCard.className = 'character-card pt-2 justify-content-center mx-auto';
+    characterCard.append(img, personName, personTitle);
+
+    const column = document.createElement('div');
+    column.appendChild(characterCard);
+    column.className = 'col mt-2 mx-0';
+    column.style.justifyContent = 'space-between';
+    column.style.paddingLeft = '0px';
+    column.style.paddingRight = '0px';
+
+    const toInsert = document.getElementById('imageContainer');
+    toInsert.appendChild(column);
+  });
+}
+
+function fetchData(callback) {
+  const url = 'https://thronesapi.com/api/v2/Characters';
+  fetch(url)
+    .then((response) => response.json())
+    .then((data) => {
+      callback(data);
+    })
+    .catch((error) => {
+      console.error('Fetch Failed', error);
+    });
```

Comment:
console.error is useful for developers and debugging. However, a casual user is likely not going to check the console dev tools.

It would be more user friendly to also display/append an error message on the web page.